pkg/globalnet/controllers/base_controllers.go

```diff
@@ -73,11 +73,21 @@ func (c *baseSyncerController) Start() error {
 	return c.resourceSyncer.Start(c.stopCh) // nolint:wrapcheck  // Let the caller wrap it
 }
 
-func (c *baseSyncerController) reconcile(client dynamic.ResourceInterface, labelSelector string,
+func (c *baseSyncerController) reconcile(client dynamic.ResourceInterface, labelSelector, fieldSelector string,
 	transform func(obj *unstructured.Unstructured) runtime.Object,
 ) {
+	listOptions := metav1.ListOptions{}
+
+	if labelSelector != "" {
+		listOptions.LabelSelector = labelSelector
+	}
+
+	if fieldSelector != "" {
+		listOptions.FieldSelector = fieldSelector
+	}
+
 	c.resourceSyncer.Reconcile(func() []runtime.Object {
-		objList, err := client.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+		objList, err := client.List(context.TODO(), listOptions)
```

Comment:
We can simplify this.

```suggestion
		objList, err := client.List(context.TODO(), metav1.ListOptions{
		    LabelSelector: labelSelector,
		    FieldSelector: fieldSelector,
		})
```